platform/iphone/gl_view.mm

```diff
@@ -284,19 +284,37 @@ - (id)initGLES {
 			kEAGLColorFormatRGBA8,
 			kEAGLDrawablePropertyColorFormat,
 			nil];
+	bool fallback_gl2 = false;
+	// Create a GL ES 3 context based on the gl driver from project settings
+	if (GLOBAL_GET("rendering/quality/driver/driver_name") == "GLES3") {
+		context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES3];
+		NSLog(@"Setting up an OpenGL ES 3 context. Based on Project Settings \"rendering/quality/driver/driver_name\"");
+		if (!context && GLOBAL_GET("rendering/quality/driver/fallback_to_gles2")) {
+			gles3_available = false;
+			fallback_gl2 = true;
+			NSLog(@"Failed to created OpenGL ES 3 context. Falling back to OpenGL ES 2.");
+		}
+	}
 
-	// Create our EAGLContext, and if successful make it current and create our framebuffer.
-	context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES3];
-
-	if (!context || ![EAGLContext setCurrentContext:context] || ![self createFramebuffer]) {
+	// Create GL ES 2 context
+	if (GLOBAL_GET("rendering/quality/driver/driver_name") == "GLES2" || fallback_gl2) {
 		context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES2];
-		gles3_available = false;
-		if (!context || ![EAGLContext setCurrentContext:context] || ![self createFramebuffer]) {
-			[self release];
+		NSLog(@"Setting up an OpenGL ES 2 context.");
+		if (!context) {
+			NSLog(@"Failed to create OpenGL ES 2 context!");
 			return nil;
 		}
 	}
 
+	if (![EAGLContext setCurrentContext:context]) {
+		NSLog(@"Failed to set EAGLContext!");
+		return nil;
+	}
+	if (![self createFramebuffer]) {
+		NSLog(@"Failed to create frame buffer!");
```

Comment:
Hopefully the verbosity of these `NSLog` statements should save us from headaches in the future.